scripts/script.js

```diff
@@ -1,118 +1,92 @@
 $(function () {
 
-    var carouselList = $('#js-myCarousel ul'),
-        interval = setInterval(changeSlideLeft, 1000),
-        i = 0,
-        $activeSlide = $('.indicators .item');
+    var carouselList = $('#js-myCarousel ul');
+    var currentSlideIndex = 0;
+    var $indicators = $('.indicators .item');
+    var lastSlideIndex = $indicators.length - 1;
+    // var interval = setInterval(function() {
+    //     moveSlidesForward(1);
+    // }, 1000);
 
 //start karuzeli//
 
-    function changeSlideLeft() {
-        if (i === 4) {
-            clearInterval(interval);
-            interval = setInterval(changeSlideRight, 1000);
+    function moveSlidesForward(numberOfSlides) {
+        if (currentSlideIndex === lastSlideIndex) {
             return;
         }
-        carouselList.animate({'margin-left': -400}, 500, moveFirstSlide);
-        i++;
-        setActiveSlide(i);
+        carouselList.animate({'margin-left': -400 * numberOfSlides}, 500, function() {
+            moveSlidesFromStart(numberOfSlides);
```

Comment:
w danej funkcji `numberOfSlides` jest parametrem ktory do tej funkcji przekazujemy. To ze akurat sie w kazdej funkcji nazywa to tylko tak zeby nam sie latwiej bylo zorientowac. Rownie dobrze mogloby byc:
```
    function moveSlidesForward(n) {
        if (currentSlideIndex === lastSlideIndex) {
            return;
        }
        carouselList.animate({'margin-left': -400 * n}, 500, function() {
            moveSlidesFromStart(n);
        });
        currentSlideIndex += n;
        setActiveSlide(currentSlideIndex);
    }

    function moveSlidesFromStart(k) {
        var firstItems = $('#js-myCarousel').find('li').slice(0, k);
        var lastItem = $('#js-myCarousel').find('li:last');
        lastItem.after(firstItems);
        carouselList.css({marginLeft: 0});
    }
```

I to bylo dzialalo tak samo. Nazwa parametru jest dostepna tylko wewnatrz funkcji.